### PR TITLE
0 sized masks are now possible

### PR DESCRIPTION
**Pull request by augusto2112**
Issue: https://github.com/pygame/pygame/issues/491

### CODE REVIEW

**Comment by illume** [comment]
Looks like a few builds are failing. You can see them by clicking the "Details" link next to Appveyor and TravisCI.

Here's one issue on a 32bit python: https://ci.appveyor.com/project/pygame/pygame/build/1.0.584/job/wi2o2m3smiuim2wq
```
======================================================================
ERROR: test_zero_size_from_surface (pygame.tests.mask_test.MaskModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\pygame\tests\mask_test.py", line 460, in test_zero_size_from_surface
    zero_w_mask = pygame.mask.from_surface(pygame.Surface((0, 100)))
ValueError: Cannot create mask with negative size
```

File: src_c/mask.c [hunk 1]
```diff
@@ -467,6 +470,11 @@ static PyObject* mask_from_surface(PyObject* self, PyObject* args)
         return NULL;
     }
 
+    if (surf->w < 0 || surf->h < 0) {
```
Comment (illume) [inline src_c/mask.c]: This check should be below `surf =`.

File: src_c/bitmask.c [hunk 2]
```diff
@@ -88,9 +88,18 @@ static INLINE unsigned int bitcount(BITMASK_W n)
 bitmask_t *bitmask_create(int w, int h)
 {
   bitmask_t *temp;
-  temp = malloc(offsetof(bitmask_t,bits) + h*((w - 1)/BITMASK_W_LEN + 1)*sizeof(BITMASK_W));
-  if (!temp)
+  size_t size;
+
+  size = offsetof(bitmask_t,bits);
+  if (w && h) {
```
Comment (illume) [inline src_c/bitmask.c]: For the case where w<=0, or h<=0 this check would pass and the size would be a weird amount.  But I guess we check that in the mask.c.

**Comment by augusto2112** [comment]
@illume I'll do it tomorrow though because today I won't have enough time. Or you prefer, you can accept it and tomorrow I submit a new PR with the tests.

**Comment by augusto2112** [comment]
@illume I added a bunch of tests. I'm not sure what should happen when trying to scale a (0, 100) mask by (0, 2) for example. Currently it stays (0, 100).

File: test/mask_test.py [hunk 3]
```diff
@@ -452,10 +458,167 @@ def test_from_threshold(self):
...
+            mask = pygame.mask.Mask(size)
+            mask.invert()
+            self.assertEqual(mask.count(), 0)
 
+    def test_zero_mask_scale(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask.scale((2, 3))
```
Comment (illume) [inline test/mask_test.py]: mask.scale should return a new mask with the requested size.
https://www.pygame.org/docs/ref/mask.html#pygame.mask.Mask.scale

So the test should be something like...
```python
scaled_mask = mask.scale((2, 3))
self.assertEqual(scaled_mask.get_size(), (2, 3))
```

**Pull request merged**